dill/_dill.py

```diff
@@ -1665,60 +1687,78 @@ def save_classmethod(pickler, obj):
         orig_func = obj.__get__(None, object)
         if isinstance(obj, classmethod):
             orig_func = getattr(orig_func, im_func) # Unbind
-    pickler.save_reduce(type(obj), (orig_func,), obj=obj)
+
+    if PY3:
+        if type(obj.__dict__) is dict:
+            if obj.__dict__:
+                state = obj.__dict__
+            else:
+                state = None
+        else:
+            state = (None, {'__dict__', obj.__dict__})
+    else:
+        state = None
+
+    pickler.save_reduce(type(obj), (orig_func,), state, obj=obj)
     log.info("# Cm")
 
 @register(FunctionType)
 def save_function(pickler, obj):
+    # obj_recursive = id(obj) in getattr(pickler, '_postproc', ())
+    # assert not obj_recursive, '%s has a bizarre structure. Open an issue.' % (obj,)
```

Comment:
Ha. That's a funny message... however, it should probably be changed to something that is less likely to offend a user.